### PR TITLE
API key interceptor

### PR DESCRIPTION
**Pull request by bychkovdmitry**
### Description

Introduced an interceptor to append `api_key` query parameter to every request URL.

### Checklist

- [X] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [X] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [X] Reviewers added

### CODE REVIEW

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api/interceptor/ApiKeyInterceptor.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+            .addQueryParameter(API_KEY, apiKey)
+            .build()
+        val request = original.newBuilder()
+            .url(url)
+            .build()
+        return chain.proceed(request)
+    }
+
+    companion object {
+        internal const val API_KEY = "api_key"
```
Comment (leandroBorgesFerreira) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api/interceptor/ApiKeyInterceptor.kt:19]: why not private?

File: stream-chat-android-client/src/test/java/io/getstream/chat/android/client/api/TokenAuthInterceptorTests.kt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package io.getstream.chat.android.client.api
 
 import io.getstream.chat.android.client.api.FakeResponse.Body
+import io.getstream.chat.android.client.api.interceptor.TokenAuthInterceptor
```
Comment (zsmb13) [inline stream-chat-android-client/src/test/java/io/getstream/chat/android/client/api/TokenAuthInterceptorTests.kt:4]: Here, the test package should be updated instead

**Review by zsmb13 — APPROVED** [review]
LGTM, one nitpick

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api/interceptor/ApiKeyInterceptor.kt (hunk 1 above)
Comment (bychkovdmitry) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api/interceptor/ApiKeyInterceptor.kt]: Yep, updated to private

File: stream-chat-android-client/src/test/java/io/getstream/chat/android/client/api/TokenAuthInterceptorTests.kt (hunk 2 above)
Comment (bychkovdmitry) [reply stream-chat-android-client/src/test/java/io/getstream/chat/android/client/api/TokenAuthInterceptorTests.kt]: Nice catch

**Pull request merged**